templates/openwisp2/settings.py

```diff
@@ -494,3 +494,37 @@
 {{ instruction }}
 
 {% endfor %}
+
+{% if django_cors.enabled %}
+
+from corsheaders.defaults import default_methods, default_headers
+
+INSTALLED_APPS.append('corsheaders')
+MIDDLEWARE.insert(2, 'corsheaders.middleware.CorsMiddleware')
+{% if django_cors.replace_https_referer %}
+MIDDLEWARE.insert(6, 'corsheaders.middleware.CorsPostCsrfMiddleware')
```

Comment:
Acc to `django-cors-headers` : `CorsMiddleware` should be placed as high as possible, especially before any middleware that can generate responses, such as `Django’s CommonMiddleware.`